drivers/hmis/spec/system/hmis/assessment_definitions_spec.rb

```diff
@@ -124,55 +175,120 @@
       click_link 'Intake'
     end
 
+    def select_member(client)
+      find('button[role="tab"]', text: client.brief_name).click
+      expect(find('button[role="tab"][aria-selected="true"]', text: client.brief_name)).to be_present
+    end
+
+    def unlock_household_assessment
+      click_button 'Unlock Assessment'
+      assert_no_text 'This assessment has been submitted.' # Unlock succeeded
+    end
+
+    def submit_household_assessment
+      click_button 'Save & Submit'
+      assert_text 'This assessment has been submitted' # Submit succeeded
+    end
+
+    def save_household_assessment
+      click_button 'Save Assessment'
+      assert_text(/Last saved [0-9] seconds? ago/) # Save succeeded
+    end
+
     context '[e1] for member with Intake that was submitted using an old form' do
+      before(:each) { select_member(c1) }
+
       it 'opens the assessment with the old form version, and upgrades to new version on unlock' do
         # expect old form
         assert_text old_definition.title
-        assert_text 'Text on old form'
         assert_text 'Entry Date'
         assert_text e1.entry_date.strftime('%m/%d/%Y')
 
-        assert_text 'Last submitted'
-        click_button('Unlock Assessment', match: :first)
-        assert_text 'Save & Submit' # Unlock succeeded
+        unlock_household_assessment
 
         # expect new form
         assert_text definition.title
+        assert_text 'New question'
         assert_no_text old_definition.title
-        assert_no_text 'Text on old form'
         expect(page).to have_field('Entry Date', with: e1.entry_date.strftime('%m/%d/%Y'))
       end
+
+      it 'submits new form fields' do
+        assert_text old_definition.title
+        unlock_household_assessment
+
+        # fill in the new question and submit, to ensure it is saved
+        fill_in 'new_question', with: 'Answer to new question'
+        expect do
+          submit_household_assessment
+        end.to change(e1.custom_assessments, :count).by(0).
+          and change(e1_assessment.custom_data_elements, :count).by(1)
+
+        expect(e1_assessment.reload.form_processor.definition_id).to eq(definition.id)
+        expect(e1_assessment.custom_data_elements.where(value_string: 'Answer to new question')).to exist
+      end
     end
 
     context '[e2] for member with Intake that was submitted using the new form' do
-      it 'opens with published form' do
-        find('button[role="tab"]', text: c2.brief_name).click
+      before(:each) { select_member(c2) }
 
+      it 'opens with published form' do
         assert_text definition.title
-        assert_no_text 'Text on old form'
         assert_text 'Entry Date'
         assert_text e2.entry_date.strftime('%m/%d/%Y')
+        assert_text 'New question'
+      end
+
+      it 'submits with published form' do
+        unlock_household_assessment
```

Comment:
I am having some flakiness issues with this one. It appears like the button was clicked, but nothing happened (sometimes)

Reply:
When I was working on the form builder spec I also encountered this. For some reason, `find("button", text: 'Unlock Assessment').trigger('click')` worked much more reliably for me than `click_button 'Unlock Assessment'`.

Reply:
Thanks, I think that helped!